File: 1164.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n, x;
		n = sc.nextInt();
		int[] teste = new int[n];
		int[] soma = new int[n];
		for (x = 0; x < n; x++) {
			soma[x] = 0;
			teste[x] = sc.nextInt();
			for (int i = 1; i < teste[x]; i++) {
				if (teste[x] % i == 0) {
					soma[x] = soma[x] + i;
				}
			}
			
		}

		for (x = 0; x < n; x++) {
			if (soma[x] == teste[x]) {
				System.out.printf("%d eh perfeito%n", teste[x]);
			} else {
				System.out.printf("%d nao eh perfeito%n", teste[x]);
			}
		}
	}
}

File: 1164.py
n = input()
for x in range(0, n, 1):
	soma = 0
	teste = input()
	for i in range(1, teste, 1):
		if (teste % i == 0):
			soma = soma + i
	if (soma == teste and teste != 0):
		print "%d eh perfeito"%(teste)
	else:
		print "%d nao eh perfeito"%(teste)
	soma = 0

